File: src/main/java/com/mxgraph/util/mxImage.java
/**
 * Copyright (c) 2007, Gaudenz Alder
 */
package com.mxgraph.util;

import java.io.Serializable;

/**
 * Implements a 2-dimensional point with double precision coordinates.
 */
public class mxImage implements Serializable, Cloneable
{

	/**
	 * 
	 */
	private static final long serialVersionUID = 8541229679513497585L;

	/**
	 * Holds the path or URL for the image.
	 */
	protected String src;

	/**
	 * Holds the image width and height.
	 */
	protected int width, height;

	/**
	 * Constructs a new point at (0, 0).
	 */
	public mxImage(String src, int width, int height)
	{
		this.src = src;
		this.width = width;
		this.height = height;
	}

	/**
	 * @return the src
	 */
	public String getSrc()
	{
		return src;
	}

	/**
	 * @param src the src to set
	 */
	public void setSrc(String src)
	{
		this.src = src;
	}

	/**
	 * @return the width
	 */
	public int getWidth()
	{
		return width;
	}

	/**
	 * @param width the width to set
	 */
	public void setWidth(int width)
	{
		this.width = width;
	}

	/**
	 * @return the height
	 */
	public int getHeight()
	{
		return height;
	}

	/**
	 * @param height the height to set
	 */
	public void setHeight(int height)
	{
		this.height = height;
	}

}

File: src/main/mxgraph/util/mxImage.js
/**
 * Copyright (c) 2006-2015, JGraph Ltd
 * Copyright (c) 2006-2015, Gaudenz Alder
 */
/**
 * Class: mxImage
 *
 * Contains the URL, width, height and optional position of an image.
 * 
 * Constructor: mxImage
 * 
 * Constructs a new image.
 */
function mxImage(src, width, height, x, y)
{
	this.src = src;
	this.width = (width != null) ? width : this.width;
	this.height = (height != null) ? height : this.height;
	this.x = (x != null) ? x : this.x;
	this.y = (y != null) ? y : this.y;
};

/**
 * Variable: src
 *
 * String that specifies the URL of the image.
 */
mxImage.prototype.src = null;

/**
 * Variable: width
 *
 * Integer that specifies the width of the image.
 */
mxImage.prototype.width = 0;

/**
 * Variable: height
 *
 * Integer that specifies the height of the image.
 */
mxImage.prototype.height = 0;

/**
 * Variable: x
 *
 * Integer that specifies the x-coordinate of the image.
 */
 mxImage.prototype.x = 0;

 /**
  * Variable: y
  *
  * Integer that specifies the y-coordinate of the image.
  */
 mxImage.prototype.y = 0;
 